### PR TITLE
Store BazelProcess output inside BazelProcessResult

### LINKED ISSUE #150: BazelProcess.waitAndGetResult() always returns empty stdout and stderr
This happens because the contents of these streams are [drained](https://github.com/JetBrains/bazel-bsp/blob/8773ecd475a658c20d609f6efc08619cd73d8173/bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/BazelProcess.java#L28) by the logging method.

### PR DESCRIPTION
**Pull request by odisseus**
Fixes #150.

### CODE REVIEW

**Review by abrams27 — COMMENTED** [review]
Laziness is cool, but I feel that here we can just drain the Stream in the constructor and just store `List<String>` in the class

**Comment by abrams27** [comment]
also please change `dev` branch to `master` - we are getting rid of a `dev` branch

File: bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/data/BazelProcessResult.java [hunk 1]
```diff
@@ -25,11 +28,35 @@ public StatusCode getStatusCode() {
...
+    return stdout.getValue().get();
   }
 
   public String getJoinedStderr() {
-    List<String> lines = BazelStreamReader.drainStream(stderr);
+    List<String> lines = stderr.getValue().get();
     return String.join(LINES_DELIMITER, lines);
   }
+
+  private static class StreamStorage extends Lazy<List<String>> {
```
Comment (lukaszwawrzyk) [inline bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/data/BazelProcessResult.java:39]: for similar use case I used Suppliers.memoize.

```
private Supplier<String> memoizedOutput(InputStream input) {
  return Suppliers.memoize(() -> BazelStreamReader.drainStream(input));
}

```

**Comment by lukaszwawrzyk** [comment]
> Laziness is cool, but I feel that here we can just drain the Stream in the constructor and just store `List<String>` in the class

Laziness might be useful, if we don't care about the output (and sometimes we don't), there is no point to do the IO to read it.

**Comment by odisseus** [comment]
See #154.

**Pull request closed without merging**